doc/cookbook/time_windows.md

```diff
@@ -0,0 +1,189 @@
+    └── 01-02-17.csv
+```
+
+As more and more daily data is added, you will end up with a directory structure that looks like:
+
+```
+binned_sales
+├── 01-01-17_to_01-03-17
+|   ├── 01-01-17.csv
+|   ├── 01-02-17.csv
+|   └── 01-03-17.csv
+├── 01-02-17_to_01-04-17
+|   ├── 01-02-17.csv
+|   ├── 01-03-17.csv
+|   └── 01-04-17.csv
+├── 01-03-17_to_01-05-17
+|   ├── 01-03-17.csv
+|   ├── 01-04-17.csv
+|   └── 01-05-17.csv
+└── etc...
+```
+
+Your second pipeline can then process these bins in parallel, via a glob pattern of `/*`, or in any other relevant way as discussed further in the ["Fixed time windows" section](#fixed-time-window-directory-structures).  Both your first and second pipelines can be easily parallelized.
+
+**Note** - When looking at the above directory structure, it may seem like there is an uneccessary duplication of the data.  However, under the hood Pachyderm deduplicates all of these files and maintains a space efficient representation of your data.  The binning of the data is merely a structural re-arrangement to allow you to process these types of rolling time windows.  
+
+**Note** - It might also seem as if there is unecessary data transfers over the network to perform the above binning.  However, the Pachyderm team is currently working on enhancements to ensure that performing these types of "shuffles" doesn't actually require transferring data over the network.
+
+## Shuffled moving time windows
+
+The advantage of the ["Rolling time window" pattern](#rolling-time-window-directory-structures) above is that both the binning and time-windowed processing can be easily parallelized.  However, you do need to put in some logic to maintain the binning directory structure.  However, there is another pattern for moving time windows that avoids the binning of the above approach and maintains an up-to-date version of a moving time-windowed data set.  The is also a two stage process:
+
+- *Pipeline 1* - Read in data, determine which files belong in your moving time window, and write the relevant files into an updated version of the moving time-windowed data set.  
+
+- *Pipeline 2* - Read in and analyze the moving time-windowed data set.
+
+**Note** - The disadvantage of this method is that "Pipeline 1" does not parallelize.  You would need to use a glob pattern of `/` in this pipeline to be able to extract the relevant files (corresponding to the moving time window) from all of the files that are input.
+
+Let's utilize our sales example again to see how this would work.  In the example, we want to keep a moving time window of the last three days worth of data.  Now say that our daily `sales` repo looks like the following:
+
+```
+sales
+├── 01-01-17.csv
+├── 01-02-17.csv
+├── 01-03-17.csv
+└── 01-04-17.csv
+```
+
+Our first pipeline would extract the last three days worth of data and put it into a repo corresponding to our moving time window (think of this as a "shuffle" step):
+
+```
+moving_sales_window
```

Comment:
Other tiny nit picks on formatting of files.

1. Does it make sense to have a `/` before or after directory names such as `sales/`? Idk what the standards around this are, just a thought. 
2. Should we use .json instead of .csv for these examples. I know is super minor difference, but dealing with csv headers is a bit annoying in Pachyderm still (you need it as a separate file or something) so maybe we don;t want to give a wrong impression and we can make our examples have json example files.